Eevee.Sleep.Bot/Workers/InGameAnnouncementCrawlingWorker.cs

```diff
@@ -0,0 +1,86 @@
+using AngleSharp.Common;
+using Discord.WebSocket;
+using Eevee.Sleep.Bot.Controllers.Mongo;
+using Eevee.Sleep.Bot.Enums;
+using Eevee.Sleep.Bot.Exceptions;
+using Eevee.Sleep.Bot.Extensions;
+using Eevee.Sleep.Bot.Models;
+using Eevee.Sleep.Bot.Utils;
+using Eevee.Sleep.Bot.Workers.Scrapers.InGameAnnouncement;
+
+namespace Eevee.Sleep.Bot.Workers;
+public class InGameAnnouncementCrawlingWorker(
+        DiscordSocketClient client,
+        ILogger<InGameAnnouncementCrawlingWorker> logger
+    ) : BackgroundService {
+    private readonly DiscordSocketClient _client = client;
+    private readonly ILogger<InGameAnnouncementCrawlingWorker> _logger = logger;
+    private readonly CancellationTokenSource _cancellationTokenSource = new();
+
+    private static readonly TimeSpan CheckInterval = TimeSpan.FromSeconds(60);
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken) {
+        _logger.LogInformation("InGameAnnouncementUpdateCheckWorker is starting.");
+        cancellationToken.Register(() => _logger.LogInformation("InGameAnnouncementUpdateCheckWorker background task is stopping."));
```

Comment:
For log message: `Stopping in-game announcement update watcher: cancellation token received`